app/src/main/java/cm/aptoide/pt/home/HomePresenter.java

```diff
@@ -704,9 +708,43 @@ private void handleLoadMoreError() {
                   .getCardId());
         })
         .map(HomeEvent::getBundle)
-        .cast(ActionBundle.class)
-        .flatMapCompletable(bundle -> home.setupAppComingSoonNotification(bundle.getActionItem()
-            .getUrl()))
+        .cast(AppComingSoonPromotionalBundle.class)
+        .flatMap(bundle -> home.setupAppComingSoonNotification(bundle.getActionItem()
+            .getUrl())
+            .andThen(Observable.just(bundle)))
```

Comment:
This is minor and up 2 you but I would change this to not force an observable creation with bundle emission when it looks like you want to actually map the emission back to bundle.

```suggestion
        .flatMap(bundle -> home.setupAppComingSoonNotification(bundle.getActionItem()
            .getUrl())
            .toObservable().map(__ -> bundle))
```

We actually want to subscribe to a fire and forget completable without changing the object... like a doOnNext that takes a function... But I don't even know the best way to do that. doOnNext with a completable and .subscribe there is just calling for garbage collection problems... dunno :hammer:

Reply:
Tested this and as we spoke on the office, it did not work. The setupNotification method is completable, so it is a completed observable, meaning we can not use doOnNext after, which is what we need.